R/notin.R

```diff
@@ -0,0 +1,9 @@
+"%notin%" = function(x, table) {
+  if (is.character(x) && is.character((table))) {
+    result = .Call(Cchin, x, table)
+  } else {
+    result = match(x, table, nomatch = 0) > 0
```

Comment:
look at print(`%in%`)

Reply:
Oh :-)  So a direct `%in%` could be submitted to R which used half as much memory. Probably just a new flag to be passed to the internal C function.  Even when the data isn't big, saving half as many small objects on the heap should be beneficial to save garbage collection. Albeit, probably most usage of `x %in% y` is a length-1 `x` though and so that's likely using R's internal global true/false off heap, and global constant small integers. But outside those cases it would be beneficial. Shouldn't be any downsides really, other than actually getting it done.

For now, this `%notin%` could drop the `>0` part to save that allocation, and that could be dealt with by the `negate` wrapper by reference.

Reply:
Thanks for taking a look @mattdowle ! I'm not sure which approach you're advocating for in your comment `%in%`. It might just be that I don't understand the R internals that well. Could you elaborate? As for clobbering the internal global TRUE/FALSE constant,  when I ran `1 %notin% 1:10` twice with this approach I would get `FALSE` the first time and `TRUE` the second time. Thank you for explaining why that was happening. My new approach solves this.